File: src/main/java/com/m2t/hachikodictionary/model/Quiz.java
package com.m2t.hachikodictionary.model;


import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Document("quizzes")
public class Quiz {

    @Id
    private String id;

    private Account account;

    private int correctAnswers;
    private int incorrectAnswers;
    private int notAnswered;
    private String difficulty;
}

File: src/main/java/com/m2t/hachikodictionary/model/Quiz.kt
package com.m2t.hachikodictionary.model

import jakarta.persistence.Entity
import jakarta.persistence.FetchType
import jakarta.persistence.GeneratedValue
import jakarta.persistence.Id
import jakarta.persistence.JoinColumn
import jakarta.persistence.ManyToOne
import jakarta.persistence.Table
import org.hibernate.annotations.GenericGenerator

@Entity
@Table(name= "quizzes", schema = "public")
data class Quiz(
    @Id
    @GeneratedValue(generator = "UUID")
    @GenericGenerator(name = "UUID", strategy = "org.hibernate.id.UUIDGenerator")
    val id: String?,

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "account_id", nullable = false)
    var account: Account,
    var correctAnswers: Int,
    var incorrectAnswers: Int,
    var notAnswered: Int = 0,
    var difficulty: String
) {
}